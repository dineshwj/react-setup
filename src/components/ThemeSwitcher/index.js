import React, { Component, Fragment } from "react";

/**
 * GenerThemeSwitcheral component to render the option to swtich themes.
 */

class ThemeSwitcher extends Component {
  /**
   * Gets called when the user changes the option in the dropdown.
   *
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   * @param {Object} allProps All props of this Button
   */

  handleThemeSwtich(e) {
    this.props.update(this.refs.themeSelector.value);
  }

  /**
   * render method to output the dropdown.
   */
  render() {
    return (
      <Fragment>
        <div
          style={{
            background: "#F3F3F3",
            paddingTop: 5,
            paddingBottom: 5,
            paddingLeft: 10,
            paddingRight: 10,
            display: "flex",
            justifyContent: "flex-end",
            borderBottom: "1px solid #999"
          }}
        >
          <label htmlFor="themes" style={{ marginRight: 20, fontSize: 12 }}>
            Select a theme to switch branding
          </label>
          <select
            id="themes"
            style={{
              background: "#fff",
              border: "1px solid #999",
              padding: "0 10px",
              borderRadius: "2px"
            }}
            ref="themeSelector"
            onChange={this.handleThemeSwtich.bind(this)}
            value={this.props.selectedTheme}
          >
            <option value="rbs_sliver">RBS Sliver</option>
            <option value="rbs_black">RBS Black</option>
            <option value="natwest_silver">Natwest Sliver</option>
            <option value="natwest_black">Natwest Black</option>
          </select>
        </div>
        {this.props.children}
      </Fragment>
    );
  }
}

export { ThemeSwitcher };
