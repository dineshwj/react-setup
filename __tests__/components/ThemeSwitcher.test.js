import React from "react";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ThemeSwitcher } from "../../src/components/ThemeSwitcher";
import toJSON from "enzyme-to-json";

configure({ adapter: new Adapter() });

let wrapper;

describe("<ThemeSwitcher />", () => {
  it("should render ThemeSwitcher", () => {
    const props = {
      dispatch: jest.fn()
    };
    wrapper = shallow(<ThemeSwitcher {...props} />);
    expect(wrapper.find("select").length).toBe(1);
    expect(wrapper.find("label").length).toBe(1);
  });

  it("should match the snapshot", () => {
    const props = {
      dispatch: jest.fn()
    };
    const tree = shallow(<ThemeSwitcher {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it("should update the state on select box change", () => {
    jest.spyOn(ThemeSwitcher.prototype, "handleThemeSwtich");
    const props = {
      dispatch: jest.fn(),
      update: jest.fn()
    };
    wrapper = mount(<ThemeSwitcher {...props} />);

    wrapper
      .find("select")
      .at(0)
      .props()
      .onChange({ target: { value: 20 } });
    expect(ThemeSwitcher.prototype.handleThemeSwtich.mock.calls.length).toBe(1);
  });
});
