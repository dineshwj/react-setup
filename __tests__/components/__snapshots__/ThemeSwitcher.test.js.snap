// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSwitcher /> should match the snapshot 1`] = `
<React.Fragment>
  <div
    style={
      Object {
        "background": "#F3F3F3",
        "borderBottom": "1px solid #999",
        "display": "flex",
        "justifyContent": "flex-end",
        "paddingBottom": 5,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 5,
      }
    }
  >
    <label
      htmlFor="themes"
      style={
        Object {
          "fontSize": 12,
          "marginRight": 20,
        }
      }
    >
      Select a theme to switch branding
    </label>
    <select
      id="themes"
      onChange={[Function]}
      style={
        Object {
          "background": "#fff",
          "border": "1px solid #999",
          "borderRadius": "2px",
          "padding": "0 10px",
        }
      }
    >
      <option
        value="rbs_sliver"
      >
        RBS Sliver
      </option>
      <option
        value="rbs_black"
      >
        RBS Black
      </option>
      <option
        value="natwest_silver"
      >
        Natwest Sliver
      </option>
      <option
        value="natwest_black"
      >
        Natwest Black
      </option>
    </select>
  </div>
</React.Fragment>
`;
